### PR TITLE
Idempotent configuration binding and support for health checks

### PR DESCRIPTION
**Pull request by autodidaddict**
0.15.x requires that all capability providers:
* Gracefully ignore attempts at binding actors that are already bound
* Support returning a health status in response to a health check

### CODE REVIEW

File: http-server/src/lib.rs [hunk 1]
```diff
@@ -23,12 +23,12 @@ use std::collections::HashMap;
...
 use std::sync::Arc;
 use std::sync::RwLock;
-use wascc_codec::core::{OP_BIND_ACTOR, OP_REMOVE_ACTOR};
+use wascc_codec::core::{OP_BIND_ACTOR, OP_REMOVE_ACTOR, OP_HEALTH_REQUEST};
 use wascc_codec::{deserialize, serialize};
 
 const CAPABILITY_ID: &str = "wascc:http_server";
 const VERSION: &str = env!("CARGO_PKG_VERSION");
-const REVISION: u32 = 2; // Increment for each crates publish
+const REVISION: u32 = 3; // Increment for each crates publish
```
Comment (brooksmtownsend) [inline http-server/src/lib.rs:31]: Doing some thinking on version v.s. revision, what purpose does revision serve if we properly use semver? I'd imagine each crates publish would correspond with a new semver, even if it's just a patch version.

**Review by brooksmtownsend — APPROVED** [review]
All looks good to me, looking for discussion on revision vs version

File: http-server/src/lib.rs (hunk 1 above)
Comment (autodidaddict) [reply http-server/src/lib.rs:31]: There's a couple things going on here:

1. We shouldn't be maintaining revision inside the code anymore since providers now have claims. In fact, we're no longer even invoking the get descriptor function
1. Revision is used to provide a simple comparison operator "is this file newer than this other file?", this gives us the ability to not have to deal with parsing semver or dealing with labels that end in `-please-cthulhu-work-this-time`. 

tl;dr when we support hot-swapping, we'll require that the revision of the new file is greater than the revision of the old file, etc.

Comment (autodidaddict) [reply http-server/src/lib.rs:31]: I'm pretty sure both Android and iOS SDKs use this pattern, where they have a human-readable semver string and a monotonically increasing revision number for discrete comparison.

Comment (brooksmtownsend) [reply http-server/src/lib.rs:31]: Okay that makes sense, I see the value in having both now. I do remember that from the iOS days, you had to be very careful to remember to bump your revision or your entire build was moot (since it wouldn't be accepted).

I would like to see with hot-swapping the ability to go back in revision, so that we can support rollbacks. I suppose you could sign the old version with a new revision, curious on your thoughts there.

Comment (autodidaddict) [reply http-server/src/lib.rs:31]: we could just make the rule that you can't hot swap revision `n` with revision `n`, and all other swaps (up or down) are allowed. The main reason behind blocking in-place hot swaps is as a security measure to prevent tampered files from being injected.

**Comment by autodidaddict** [comment]
Added idempotency and health check upgrades for redis and NATS. Going to create an issue for the other providers to all be upgraded to generated code, health checks, idempotency, and the stop function.
